remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerInterceptor.java

```diff
@@ -337,66 +369,81 @@ public Object visitPutMapCommand(InvocationContext ctx, PutMapCommand command) {
          }
 
          if (progressCallback != null) {
-            updateGlobalErrors(rCtx, progressCallback.getErrorFiles(), flagsBitSet);
+            CompletionStage<Void> schemaUpdate = updateSchema(rCtx, progressCallback);
+            CompletionStage<Object> errorUpdate = updateGlobalErrors(rCtx, progressCallback.getErrorFiles().keySet(), flagsBitSet);
+
+            return asyncValue(CompletionStages.allOf(schemaUpdate, errorUpdate));
          }
+         return InvocationStage.completedNullStage();
       });
    }
 
    @Override
    public Object visitRemoveCommand(InvocationContext ctx, RemoveCommand command) {
-      if (ctx.isOriginLocal()) {
-         if (!(command.getKey() instanceof String)) {
-            throw log.keyMustBeString(command.getKey().getClass());
+      if (!ctx.isOriginLocal()) {
+         return invokeNext(ctx, command);
+      }
+      if (!(command.getKey() instanceof String)) {
+         throw log.keyMustBeString(command.getKey().getClass());
+      }
+      String key = (String) command.getKey();
+      if (!shouldIntercept(key)) {
+         return invokeNext(ctx, command);
+      }
+      // lock .errors key
```

Comment:
I suggest replacing all the instances of the "lock .errors key" comment with "lock global errors key", the fact that `.errors` is the key for global errors isn't obvious.

Maybe even extract a method `lockGlobalErrors` to be used everywhere.

Reply:
This isn't really related to my changes, other than indentation. I can make the text change as I don't want to mess something up while I am gone.

Reply:
Yeah, it's not your comment, it was just annoying me every time I read it :)